Show correct XP info (#2)

pledbrook opened the pull request:
The XP information now delegates to `XCGS_Unit.GetTotalNumKills()` for maximum compatibility with other mods.

Fixes #2.

pledbrook committed:
Show correct XP info (#2)

The XP information now delegates to `XCGS_Unit.GetTotalNumKills()` for
maximum compatibility with other mods.

Fixes #2.

bountygiver commented:
This breaks the more granular assist tracking, please make this functionality optional by checking for a config option

bountygiver reviewed (changes requested):
Put this functionality behind a config check

pledbrook committed:
Support both kill XP and kill assist equivalent

This restores the original behaviour of Detailed Soldier Lists by
showing kill assist equivalent by default, but adds a config option to
display normal kill XP instead.

pledbrook commented:
Both kill XP and kill assist equivalent should now display properly and honour any customisations made by other mods.

bountygiver commented on DetailedSoldierListWOTC/Src/DetailedSoldierListWOTC/Classes/UIPersonnel_SoldierListItemDetailed.uc:
```diff
@@ -63,7 +65,18 @@ simulated function string GetPromotionProgress(XComGameState_Unit Unit)
+	{
+		// Deduct the existing assists contribution before multiplying by KillAssistsPerKill.
+		// Ignoring psi credits as they appear not to be used at all.
+		NumKills = Unit.GetTotalNumKills() - Round(Unit.KillAssistsCount) / ClassTemplate.KillAssistsPerKill;
```
instead of deducting then adding, you can just multiply both kills and kills required by kill assist per kill then adding assisst modulo kill assist per kill

pledbrook replied:
I didn't quite follow what you were saying here, but hope the change I pushed is satisfactory.

pledbrook committed:
Change calculation for XP as assists

The calculation for kill XP as assists equivalent is a bit simpler now.

pledbrook commented:
I pushed the requested changes.
